proto/buf/alpha/registry/v1alpha1/labels.proto

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.alpha.registry.v1alpha1;
+
+import "google/protobuf/timestamp.proto";
+
+enum LabelNamespace {
+  LABEL_NAMESPACE_UNSPECIFIED = 0;
+  LABEL_NAMESPACE_TAG = 1;
+  LABEL_NAMESPACE_BRANCH = 2;
+  LABEL_NAMESPACE_GIT_COMMIT = 3;
+}
+
+message Label {
+  LabelNamespace namespace = 1;
+  string name = 2;
+  LabelValue label_value = 3;
+}
+
+message LabelValue {
+  string commit_id = 1;
+}
+
+service LabelService {
+  rpc CreateLabel(CreateLabelRequest) returns (CreateLabelResponse);
+  rpc MoveLabel(MoveLabelRequest) returns (MoveLabelResponse);
+  rpc GetLabels(GetLabelsRequest) returns (GetLabelsResponse);
+}
+
+message CreateLabelRequest {
+  Label label = 1;
+  optional string author = 2;
+  optional google.protobuf.Timestamp create_time = 3;
+}
+
+message CreateLabelResponse {
+  string commit_id = 1;
+}
+
+message MoveLabelRequest {
+  Label label = 1;
+  LabelValue from = 2;
```

Comment:
Looking at this again makes this a very interesting form. What happens when `label` contains a value that's different than `from`?

```proto
{
  "label": {"namespace": "LABEL_NAMESPACE_TAG", "name": "ponies", value: "dashie"},
  "from": "rainbow dash",
  "to": "rd"
}
```
Moving `LabelValue` out of `Label` makes it clear values are not supposed to be supplied where they don't make sense.

Reply:
The intent here would be that you can only move a label if the `from` supplied matches the label value -- this helps prevent some of the race conditions if multiple move operations are being fired. That being said, I should probably demonstrate that `from` is optional.

Reply:
What I'm raising is the label's from value is specified twice in this message. Once in the `label` field and once in the `from` field. My example exemplifies the question of what happens when the from value is duplicated but with different values?

Reply:
Ah yeah, I see. Okay, this makes sense. Let me adjust this. For `GetLabels`, I do think we may want to return `LabelValue` alongside the name and namespace, so let me push up an update :)